File: hw2/insertform.pl
#!/usr/bin/perl

use Data::Dumper;
use CGI;
my $q = CGI->new;
 
print $q->header;

print $q->start_html(-title => 'A web form');


print "<script src='form.js'    type='text/javascript'></script>\n";


print $q->start_form(
        -name    => 'main_form',
        -id      => 'myForm',
        -method  => 'POST',
        -enctype => &CGI::URL_ENCODED,
        -onsubmit => 'validate()',
        -action => 'processform.pl', # Defaults to 
                                                 # the current program
);


print "Name: ".$q->textfield(
        -name      => 'name',
        -value     => '',
        -size      => 20,
        -maxlength => 30,
    );

print $q->br;

print "Favorite Color: ".$q->textfield(
        -name      => 'color',
        -value     => '',
        -size      => 20,
        -maxlength => 30,
    );

print $q->br;

print $q->popup_menu(-name=>'ins_method',
				 -values=>['GET','POST'],
				 -default=>'POST');

print $q->br;

print $q->submit(-value=>'Submit');

print $q->end_form;

print $q->end_html;

File: hw2/insertform.php
<!DOCTYPE HTML>
<html lang="en-US">
<head>
</head>

<body>
	
<?php
	echo '<p>Hello World from PHP on ' . date('Y-m-d h:i:s a') . '</p>';


$background_color = '#CCDDEE';

echo "<body style='background-color:$background_color'>";


?>




<form id='myForm' name='main_form' method='POST' onsubmit='validate()' action ='processform.php'>

	Name: <input type="text" name="name" value="<?php echo $name;?>">
	<br/>
	Favorite Color: <input type="text" name="color" value="<?php echo $color;?>">


</form>



</body>

</html>